### PR TITLE
fix(components/lookup): show more modal infinite scroll respects only show selected checkbox

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/427/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/427/storybooks/ag-grid-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/427/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/427/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/427/playground/)

File: libs/components/lookup/src/lib/modules/lookup/lookup-show-more-modal.component.ts [hunk 1]
```diff
@@ -136,8 +136,13 @@ export class SkyLookupShowMoreModalComponent
...
     }
 
+    // if onlyShowSelected is checked, then only iterate through the initial selection
+    const items = this.items.filter(
+      (item) => item.selected || !this.onlyShowSelected
+    );
+
     this.#itemIndex = this.#itemIndex + 10;
-    this.searchItems(this.items).then((searchedItems) => {
+    this.searchItems(items).then((searchedItems) => {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/lookup/src/lib/modules/lookup/lookup-show-more-modal.component.ts:154]: Needs a unit test.

File: libs/components/lookup/src/lib/modules/lookup/lookup-show-more-modal.component.ts [hunk 2]
```diff
@@ -136,8 +136,13 @@ export class SkyLookupShowMoreModalComponent
       this.#changeDetector.markForCheck();
     }
 
+    // if onlyShowSelected is checked, then only iterate through the initial selection
+    const items = this.items.filter(
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/lookup/src/lib/modules/lookup/lookup-show-more-modal.component.ts]: I wonder if doing the filter conditionally when `this.onlyShowSelected` is true is a little more efficient. With the way this is written we will iterate through all of `this.items` every time no matter if it is needed or not. Some consumers load a crazy amount of data into `lookup` so I wonder if we should only do the `.filter` if it is needed and just set the local variable to `this.items` if not.

File: libs/components/lookup/src/lib/modules/lookup/lookup-show-more-modal.component.ts (hunk 1 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/lookup/src/lib/modules/lookup/lookup-show-more-modal.component.ts:154]: Added

File: libs/components/lookup/src/lib/modules/lookup/lookup-show-more-modal.component.ts (hunk 2 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/lookup/src/lib/modules/lookup/lookup-show-more-modal.component.ts]: Good idea. Changed

File: libs/components/lookup/src/lib/modules/lookup/lookup.component.spec.ts [hunk 3]
```diff
@@ -5010,6 +5056,52 @@ describe('Lookup component', function () {
...
+              await fixture.whenStable();
+              fixture.detectChanges();
+
+              expect(getRepeaterItemCount()).toBe(18);
+
+              (
+                document.querySelector(
+                  '.sky-lookup-show-more-modal-close'
+                ) as HTMLElement
+              )?.click();
```
Comment (Blackbaud-SteveBrush) [inline libs/components/lookup/src/lib/modules/lookup/lookup.component.spec.ts]: Can any of this duplicate code be placed in a helper function so if we need to make changes to both tests the logic lives in one place?

Comment (Blackbaud-CoreyArcher) [reply libs/components/lookup/src/lib/modules/lookup/lookup.component.spec.ts]: Added more helper methods

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**